generators/client/templates/angular/src/main/webapp/app/shared/auth/_user-route-access-service.ts

```diff
@@ -16,7 +16,13 @@ export class UserRouteAccessService implements CanActivate {
     }
 
     canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Promise<boolean> {
-        return this.checkLogin(route.data['authorities'], state.url);
+
+        const authorities = route.data['authorities'];
+        if (authorities && authorities.length == 0) {
```

Comment:
If authorities is undefined, we should return true as well.  Otherwise, another error will be thrown later in `Principal.hasAnyAuthority` when looping the array.  Maybe something like this?
```
if (!authorities || authorities.length == 0) {
```